rs/crypto/secp256k1/src/lib.rs

```diff
@@ -837,6 +907,44 @@ impl PublicKey {
         }
     }
 
+    /// BIP341 derivation
+    fn derive_bip341(&self, ttr: &[u8]) -> Result<Self, InvalidTaprootHash> {
```

Comment:
We could also check in unit tests that functions produce correct results against third-party libraries. But not necessarily in this PR.

Reply:
Differential fuzzers might also make sense

Reply:
We check already that the signatures are accepted by `bitcoin`, is there something else we should be doing?